healthcheck.php

```diff
@@ -0,0 +1,36 @@
+<?php
+namespace Automattic\VIP;
+
+use Automattic\VIP\Utils\Context;
+/**
+ * Utility Healthcheck class that runs further than cache-healthcheck and allow to verify the app code is working.
+ */
+class Healthcheck {
+	/**
+	 * By default the site is considered healthy unless filter value returns falsy value
+	 *
+	 * @var boolean
+	 */
+	protected bool $healthy = true;
+	public function __construct() {
+		$this->check();
+	}
+	public function check() {
+		$this->healthy = (bool) apply_filters( 'vip_site_healthcheck', $this->healthy );
```

Comment:
i would propose that rather than a bool, this returns a `string[]`, where empty array means no errors (aka healthy) and non-empty indicates any number of reasons for the unhealthiness - then we can echo these reasons before exiting so it is trivial to look at a pod and ask it why it thinks it is not healthy, and not have to guess.

Reply:
Agree - the conversion to bool can be done in `is_healthy()`, but a new function could return the array of errors (`get_health_errors()` or whatever).

Reply:
@bnu0 Addressed.